code/Test_definitions/quality-on-demand-extendQosSessionDuration.feature

```diff
@@ -0,0 +1,194 @@
+        When the request "extendQosSessionDuration" is sent
+        Then the response status code is 401
+        And the response header "x-correlator" has same value as the request header "x-correlator"
+        And the response header "Content-Type" is "application/json"
+        And the response property "$.status" is 401
+        And the response property "$.code" is "UNAUTHENTICATED"
+        And the response property "$.message" contains a user friendly text
+
+    # In this case both codes could make sense depending on whether the access token can be refreshed or not
+    @quality_on_demand_extendQosSessionDuration_401.2_expired_access_token
+    Scenario: Error response for expired access token
+        Given the header "Authorization" is set to an expired access token
+        And the request body is set to a valid request body
+        When the request "extendQosSessionDuration" is sent
+        Then the response status code is 401
+        And the response header "x-correlator" has same value as the request header "x-correlator"
+        And the response header "Content-Type" is "application/json"
+        And the response property "$.status" is 401
+        And the response property "$.code" is "UNAUTHENTICATED" or "AUTHENTICATION_REQUIRED"
+        And the response property "$.message" contains a user friendly text
+
+    @quality_on_demand_extendQosSessionDuration_401.3_invalid_access_token
+    Scenario: Error response for invalid access token
+        Given the header "Authorization" is set to an invalid access token
+        And the request body is set to a valid request body
+        When the request "extendQosSessionDuration" is sent
+        Then the response status code is 401
+        And the response header "x-correlator" has same value as the request header "x-correlator"
+        And the response header "Content-Type" is "application/json"
+        And the response property "$.status" is 401
+        And the response property "$.code" is "UNAUTHENTICATED"
+        And the response property "$.message" contains a user friendly text
+
+
+    # Errors 403
+
+    @quality_on_demand_extendQosSessionDuration_403.1_session_token_mismatch
+    Scenario: QoS session not created by the API client given in the access token
+        # To test this, a token have to be obtained for a different client
+        Given the header "Authorization" is set to a valid access token emitted to a client which did not created the QoS session        And the header "Authorization" is set to a valid access token emitted for a different device
+        When the request "extendQosSessionDuration" is sent
+        Then the response status code is 403
+        And the response header "x-correlator" has same value as the request header "x-correlator"
+        And the response header "Content-Type" is "application/json"
+        And the response property "$.status" is 403
+        And the response property "$.code" is "PERMISSION_DENIED"
```

Comment:
I agree that the generic "PERMISSION_DENIED" is appropriate here, but the API definitions lists also the more specific "INVALID_TOKEN_CONTEXT" for this scenario.

Reply:
INVALID_TOKEN_CONTEXT was intended for device mismatch between the access_token and the request content. In this case the provided access token may be valid but the accessed resource is not accessible for the client, so maybe PERMISSION_DENIED is more common. I can allow both and add a note to the readme and test plan